### PR TITLE
fix: Bug in sdk-client where withReasons was not passed to streamer.

### LINKED ISSUE #383: LDEvaluationReason is null, if a flag is not linked to an experiment
Hi! 👋 
      
Firstly, thanks for your work on this project! 🙂

Today I used [patch-package](https://github.com/ds300/patch-package) to patch `@launchdarkly/js-sdk-common@2.2.0` for the project I'm working on.

Here is the diff that solved my problem:

```diff
diff --git a/node_modules/@launchdarkly/js-sdk-common/dist/api/data/LDEvaluationDetail.d.ts b/node_modules/@launchdarkly/js-sdk-common/dist/api/data/LDEvaluationDetail.d.ts
index c8d689d..fe994e6 100644
--- a/node_modules/@launchdarkly/js-sdk-common/dist/api/data/LDEvaluationDetail.d.ts
+++ b/node_modules/@launchdarkly/js-sdk-common/dist/api/data/LDEvaluationDetail.d.ts
@@ -37,7 +37,8 @@ export interface LDEvaluationDetailTyped<TFlag> {
     variationIndex?: number | null;
     /**
      * An object describing the main factor that influenced the flag evaluation value.
+     * This is null if the flag is not linked to a running experiment.
      */
-    reason: LDEvaluationReason;
+    reason?: LDEvaluationReason | null;
 }
 //# sourceMappingURL=LDEvaluationDetail.d.ts.map
\ No newline at end of file
```

<em>This issue body was [partially generated by patch-package](https://github.com/ds300/patch-package/issues/296).</em>

### PR DESCRIPTION
**Pull request by yusinto**
Fixes #383.

### CODE REVIEW

File: packages/shared/sdk-client/src/LDClientImpl.test.ts [hunk 1]
```diff
@@ -161,6 +162,34 @@ describe('sdk-client object', () => {
...
+      expect.anything(),
+      '/stream/path',
+      expect.anything(),
+      undefined,
+      expect.anything(),
+    );
+  });
+
+  test('identify success withReasons', async () => {
+    const carContext: LDContext = { kind: 'car', key: 'mazda-cx7' };
```
Comment (kinyoklion) [inline packages/shared/sdk-client/src/LDClientImpl.test.ts:176]: Please use a fictional car brand here.

File: packages/shared/sdk-client/src/LDClientImpl.ts [hunk 2]
```diff
@@ -321,10 +321,14 @@ export default class LDClientImpl implements LDClient {
       }
     } else {
       this.streamer?.close();
+      let streamUri = this.createStreamUriPath(context);
+      if (this.config.withReasons) {
+        streamUri = `${streamUri}?withReasons=true`;
```
Comment (kinyoklion) [inline packages/shared/sdk-client/src/LDClientImpl.ts:326]: I am not sure about this. Generally we default to not including reasons, and only including them if the user configured the SDK to request them. (Which most people do not need, so we don't have to spend the money sending them.) That configuration then still needs to make its way to the streaming and polling configuration.

The reason does have to be optional on the evaluation detail for a client SDK though.

**Review by kinyoklion — CHANGES_REQUESTED** [review]

File: packages/shared/sdk-client/src/LDClientImpl.ts (hunk 2 above)
Comment (yusinto) [reply packages/shared/sdk-client/src/LDClientImpl.ts:326]: I understand this. Hence the check on line 325 to only include it if the user explicitly config'd the sdk to do so. Is this what you are talking about?

Comment (kinyoklion) [reply packages/shared/sdk-client/src/LDClientImpl.ts:326]: I failed at reading.

**Review by kinyoklion — APPROVED** [review]

File: packages/shared/sdk-client/src/LDClientImpl.ts (hunk 2 above)
Comment (yusinto) [reply packages/shared/sdk-client/src/LDClientImpl.ts:326]: You were right 99% the time I was wrong though. Thanks for your review!

**Pull request merged**